src/cdk-experimental/listbox/listbox.spec.ts

```diff
@@ -0,0 +1,92 @@
+import {
+  ComponentFixture,
+  async,
+  TestBed,
+} from '@angular/core/testing';
+import {Component} from '@angular/core';
+import {By} from '@angular/platform-browser';
+import {
+  CdkOption,
+  CdkListbox,
+  CdkListboxModule
+} from './index';
+import {dispatchMouseEvent} from '@angular/cdk/testing/private';
+
+describe('CdkOption', () => {
+
+  describe('selection state change', () => {
+    let fixture: ComponentFixture<CdkListboxWithCdkOptions>;
+    let listboxInstance: CdkListbox;
+    let options: CdkOption[];
+
+    beforeEach(async(() => {
+      TestBed.configureTestingModule({
+        imports: [CdkListboxModule],
+        declarations: [CdkListboxWithCdkOptions],
+      }).compileComponents();
+
+      fixture = TestBed.createComponent(CdkListboxWithCdkOptions);
+      fixture.detectChanges();
+      listboxInstance = fixture.debugElement.query(By.directive(CdkListbox))
+          .injector.get(CdkListbox);
+      options = listboxInstance._options.toArray();
+    }));
+
+    it('should generate a unique optionid for each option', () => {
+      options.forEach(option => {
+        expect(option.getOptionId()).toMatch(/cdk-option-\d+/);
+      });
+    });
+
+    it('should have set the selected input of the options to null by default', () => {
+      options.forEach(option => {
+        expect(option.selected).toBe(null);
+      });
+    });
+
+    it('should update aria-selected when selected is changed programmatically', () => {
+      expect(options[1].getElementRef().nativeElement.getAttribute('aria-selected')).toBe(null);
```

Comment:
nit: you can use `toBeNull()` 
```suggestion
      expect(options[1].getElementRef().nativeElement.getAttribute('aria-selected')).toBeNull();
```
(here and elsewhere)